src/customs/useCalculator.ts

```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react'
+
+import { DigitNumber, Operation } from '@/types/calculator'
+
+import { calculator } from '@/utils/calculator'
+
+const useCalculator = () => {
+  const [total, setTotal] = useState(0)
+  const [operator, setOperator] = useState({
+    leftPort: 0,
+    rightPort: 0,
+    expression: '',
+  })
+
+  const handleDigit = (digit: DigitNumber) => {
+    const { expression } = operator
+    const newTotal = Boolean(expression) ? digit : Number(total.toString().concat(digit.toString()))
+    if (newTotal > 999) return alert('세 자리 수 이상 입력할 수 없습니다.')
+    if (Boolean(expression)) {
+      setOperator({ ...operator, rightPort: newTotal })
```

Comment:
```suggestion
      setOperator(prev => ({ ...prev, rightPort: newTotal }))
```

이렇게 Dispatcher에 콜백함수를 넘기는 것도 좋을것 같아요. 이렇게 한다면, 

매번 operator를 구조분해 할당을 안해도 될 것 같아요..!